src/ser.rs

```diff
@@ -1304,6 +1554,14 @@ pub trait Formatter {
         dtoa::write(writer, value).map(|_| ())
     }
 
+    /// Writes a number that has already been rendered to a string.
+    #[inline]
+    fn write_number_str<W: ?Sized>(&mut self, writer: &mut W, value: &str) -> Result<()>
```

Comment:
The returned Result type should be io::Result instead of serde_json::Result to be consistent with the rest of the Formatter methods.